docarray/doc_index/backends/elastic_doc_index.py

```diff
@@ -0,0 +1,454 @@
+            super().__init__()
+            self._search = Search(
+                using=outer_instance._client, index=outer_instance._index_name, **kwargs
+            )
+
+        def build(self, *args, **kwargs) -> Any:
+            return self
+
+        def query(self, *args, **kwargs):
+            self._search = self._search.query(*args, **kwargs)
+            return self
+
+        def filter(self, *args, **kwargs):
+            self._search = self._search.filter(*args, **kwargs)
+            return self
+
+        def exclude(self, *args, **kwargs):
+            self._search = self._search.exclude(*args, **kwargs)
+            return self
+
+        def script_fields(self, **kwargs):
+            self._search = self._search.script_fields(**kwargs)
+            return self
+
+        def sort(self, *args, **kwargs):
+            self._search = self._search.sort(*args, **kwargs)
+            return self
+
+        # TODO: add more methods
+
+        find = _raise_has_substitue('find')
+        text_search = _raise_has_substitue('text_search')
+        find_batched = _raise_not_composable('find_batched')
+        filter_batched = _raise_not_composable('find_batched')
+        text_search_batched = _raise_not_composable('text_search')
+
+    def build_query(self, **kwargs) -> QueryBuilder:
+        """
+        Build a query for this DocumentIndex.
+        """
+        return self.QueryBuilder(self, **kwargs)  # type: ignore
+
+    @dataclass
+    class DBConfig(BaseDocumentIndex.DBConfig):
+        hosts: Union[str, List[str], None] = 'http://localhost:9200'
+        index_name: Optional[str] = None
+        es_config: Dict[str, Any] = field(default_factory=dict)
+        index_settings: Dict[str, Any] = field(default_factory=dict)
+
+    @dataclass
+    class RuntimeConfig(BaseDocumentIndex.RuntimeConfig):
+        default_column_config: Dict[Type, Dict[str, Any]] = field(
+            default_factory=lambda: {
+                np.ndarray: {
+                    'type': 'dense_vector',
+                    'dims': 128,
+                },
+                docarray.typing.ID: {'type': 'keyword'},
+                bool: {'type': 'boolean'},
+                int: {'type': 'integer'},
+                float: {'type': 'float'},
+                str: {'type': 'text'},
+                # `None` is not a Type, but we allow it here anyway
+                None: {},  # type: ignore
+            }
+        )
+
+    ###############################################
+    # Implementation of abstract methods          #
+    ###############################################
+
+    def python_type_to_db_type(self, python_type: Type) -> Any:
+        """Map python type to database type."""
+        for allowed_type in ELASTIC_PY_VEC_TYPES:
+            if issubclass(python_type, allowed_type):
+                return np.ndarray
+
+        if python_type in ELASTIC_PY_TYPES:
+            return python_type
```

Comment:
I am not sure this is done the way it is intended (so maybe the guidance need to be revised). What this should output is the type _as it is represented in the database_. So `docarray.typing.ID` would not be a valid type in the database, right? This should be mapped to something like 'string' or 'keyword'. I think you have put that info in the mapping above, but actually it should go here so it can automatically be parsed into `_ColumnInfo` :)